lib/pushmi_pullyu/aip.rb

```diff
@@ -0,0 +1,40 @@
+require 'pushmi_pullyu'
+require 'bagit'
+require 'archive/tar/minitar'
+
+module PushmiPullyu::AIP
+  # Exceptions
+  class BagInvalid < StandardError; end
+  class NoContentFilename < StandardError; end
+  class FedoraFetchError < StandardError; end
+  class SolrFetchError < StandardError; end
+
+  def self.create(noid, should_skip_download: false, should_clean_work_directories: true)
```

Comment:
Really good point. I should also do some validation so we don't noids like '../../etc/password'.
Is there a validator for what noid content can be? I'm guessing ^[A..Za..z0..9]+$ ... don't know if there are any other allowed characters.